compiler/noirc_evaluator/src/ssa/validation/mod.rs

```diff
@@ -148,6 +148,24 @@ impl<'f> Validator<'f> {
                     panic!("ArrayGet/ArraySet index must be u32");
                 }
             }
+            Instruction::Call { func, arguments } => {
+                if let Value::Intrinsic(intrinsic) = &dfg[*func] {
+                    let value_typ = dfg.type_of_value(arguments[0]);
+                    assert!(matches!(value_typ, Type::Numeric(NumericType::NativeField)));
+
+                    if matches!(intrinsic, Intrinsic::ToRadix(_)) {
+                        assert_eq!(arguments.len(), 2);
+                        let radix_typ = dfg.type_of_value(arguments[1]);
+                        assert!(matches!(
+                            radix_typ,
+                            Type::Numeric(NumericType::Unsigned { bit_size: 32 })
+                        ));
+                    } else {
+                        // Intrinsic::ToBits always has a set radix
+                        assert_eq!(arguments.len(), 1);
+                    }
```

Comment:
This should be a `match` rather than `if-else` so we don't hit all the other intrinsics.